api/v1alpha1/openapi.yaml

```diff
@@ -2539,10 +2539,42 @@ components:
         - type: object
           properties:
             inline:
-              type: object
-              additionalProperties: true
+              type: array
+              items:
+                $ref: '#/components/schemas/FileSpec'
           required:
           - inline
+    FileSpec:
+      type: object
+      properties:
+        path:
+          type: string
+          description: The absolute path to the file on the device. Note that any existing file will be overwritten.
+        contents:
+          type: string
+          description: The URL-encoded or base64-encoded contents of the file.
+        contentsEncoding:
+          type: string
+          description: How the contents are encoded.
+          enum:
+          - base64
+          - url
+        mode:
+          type: integer
+          description: |
+            The file’s permission mode. You may specify the more familiar octal with a leading zero (e.g., 0644) or as
+            a decimal without a leading zero (e.g., 420). Setuid/setgid/sticky bits are supported. If not specified,
+            the permission mode for files defaults to 0644.
+        user:
+          type: string
+          description: The file's owner, specified either as a name or numeric ID. Defaults to "root".
+        group:
+          type: string
+          description: The file's group, specified either as a name or numeric ID.
```

Comment:
What does this default to?

Reply:
What providers can populate user/group etc?

Reply:
Also defaults to "root", will fix.
@hexfusion what do you mean by providers?

Reply:
Sorry let me rephrase what ConfigProviders allow setting group, user, perms for ignition/Files?

Reply:
Only this one.  Git/k8s secret/HTTP don't have a way to express ownership or permissions.  Does this suggest it's not needed here?  Or that we need to supplement the others?

Reply:
Just an observation at this point I don't know the answer.

Reply:
Is it even realistically possible to add to Git/HTTP? Secrets you could consider labels or annotations possibly

Reply:
Anything is possible :)
I mean you can extend our API, for example:
```
httpRef:
  repository: myrepo
  suffix: myfile.yaml
  filePath: /etc/foo
  owner: avishay
  group: avishay
  permissions: 0o600
```

But for now @fzdarsky suggested that we ignore this difference for now.  We need permissions for inline, and can consider how to add it for other config providers later on.